### ISSUE TITLE
koroshell

### ISSUE THREAD

**Issue by Lurkios** [DeadSix27/waifu2x-converter-cpp#9]
Is there any method for getting the updated binaries/level 3 noise reduction into koroshell?

**Comment by DeadSix27** [comment]
Level 3 noise reduction is in? Or what do you mean?
```
   --noise_level <1|2|3>
     noise reduction level
```

**Comment by Lurkios** [comment]
Right. Except the koroshell UI seems to be hard coded to only accept "low" and "high" - as well as lacks options for forcing opencv, meaning if I swap the binary directly I can only run it in a VM (crashes otherwise, as the binary does if I run it with default settings) and only with level 1 and 2 noise reduction. Since koroshell was listed on the project page I was thinking there might have been some compatibility between the two. Koroshell by itself is great, but it does seem to do something wonky with RGB whereas your version here, and thank you for it, works just fine.

**Comment by DeadSix27** [comment]
@Lurkios is the koroshell UI in this repo? Like does my fork build it?

**Comment by Lurkios** [comment]
@DeadSix27 To be honest I'd have had no idea. I made this github account today and it's my first time using the site. No need to be an ass about it, a simple "no" suffices.

**Comment by DeadSix27** [comment]
@Lurkios I have no idea what "koroshell UI". If you consider not knowing something "being an ass about it" ...

**Comment by Lurkios** [comment]
@DeadSix27 I read that as dripping sarcasm. If that wasn't the case then my apologies.

In the ReadMe.md file shown on the "main page" (terminology?) of this fork there's a link to [waifu2x_win_koroshell](http://inatsuka.com/extra/koroshell/ ). It's a gui/front-end for the original version of the converter (I think) by nagadomi (https://github.com/nagadomi/waifu2x) that lets you drag&drop a file or multiple to process - not necessary but very convenient. All it really looks to be doing is passing arguments on to the binary for nagadomi's version. Dropping your binary in over the existing one, along with the models, actually does work - but because your version introduces several new options over nagadomi's those are missing from the gui (including an option I believe most nvidia users are probably going to need to get this version to work). I was thinking there was a chance that, since koroshell was mentioned right under the link to your binary as a "waifu2x converter GUI" - there might be some method of getting those additional options to work with it. It sounds like that isn't the case.

Working on my own UI for it now. No idea what I'm doing, I expect the results will be interesting.

**Comment by DeadSix27** [comment]
@Lurkios ah ye I never use a GUI i just have a python script for that. So I didn't even know that GUI existed.
The koroshell UI doesn't support level 3 yes, it's also not part of this source codes so I can not change that in anyway. 
There is https://github.com/DeadSix27/waifu2x-converter-cpp/tree/master/w32-apps
But I never touched that, I'm waiting on OpenCV to implement support for unicode image file names until then its unlikely Ill make a UI or something, feel free to do one yourself tho.

**Comment by YukihoAA** [comment]
@Lurkios SnowShell will be help for you! https://github.com/YukihoAA/waifu2x_snowshell/releases/tag/v1.1
